run_unit_tests.sh

```diff
@@ -32,35 +39,42 @@ if [ "$RUN_GPU_TESTS" == "1" ]; then
     SHARED_OPTIONS+=" --test_env RUN_GPU_TESTS=1"
 fi
 
-test_success_procedure() {
-    grep -a " ms \| ms)" ${TEST_LOG}
-    tail -50 ${TEST_LOG}
-}
+compress_logs() {
+    tar -czf test_logs.tar.gz ${TEST_LOG}
+    rm -rf ${TEST_LOG}
+    rm -rf tmp.log
+} 
+
 generate_coverage_report() {
     test_success_procedure
     genhtml --output genhtml "$(bazel info output_path)/_coverage/_coverage_report.dat"
 }
-test_fail_procedure() {
-    test_success_procedure
-    cat ${TEST_LOG} && rm -rf ${TEST_LOG} && exit 1
-}
+
 echo "Run test: ${RUN_TESTS}"
 echo "Run GPU test: ${RUN_GPU_TESTS}"
 echo "Run coverage: ${CHECK_COVERAGE}"
 if [ "$RUN_TESTS" == "1" ] ; then
     if [ "$CHECK_COVERAGE" == "1" ] ; then
         { bazel coverage --instrumentation_filter="-src/test" --combined_report=lcov \
             ${SHARED_OPTIONS} ${TEST_FILTER} \
-            //src:ovms_test ${BAZEL_OPTIONS} > ${TEST_LOG} 2>&1 || \
-            test_fail_procedure; } && \
+            //src:ovms_test ${debug_bazel_flags} > ${TEST_LOG} 2>&1 || \
+            compress_logs && exit 1; } && \
             generate_coverage_report;
     fi
-    { 
-        bazel test --jobs=$JOBS ${BAZEL_OPTIONS} ${SHARED_OPTIONS} "${TEST_FILTER}" //src/python/binding:test_python_binding && \
-        bazel test \
-          ${SHARED_OPTIONS} "${TEST_FILTER}" \
-          //src:ovms_test ${BAZEL_OPTIONS} > ${TEST_LOG} 2>&1 || \
-          test_fail_procedure; } && \
-          test_success_procedure && \
-          rm -rf ${TEST_LOG};
+    bazel build --jobs=$JOBS ${debug_bazel_flags} //src:ovms_test 
+    set +x
+    echo "Executing unit tests"
+    failed=0
+    for i in `./bazel-bin/src/ovms_test --gtest_list_tests --gtest_filter="-LLMChatTemplateTest.*:LLMOptionsHttpTest.*" | grep -vE '^ ' | cut -d. -f1` ; do
+        ./bazel-bin/src/ovms_test --gtest_filter="$i.*" > tmp.log 2>&1 || ( failed=1 ; echo $i ; cat tmp.log ) 
+        cat tmp.log >> ${TEST_LOG}
+    done
+    for i in `./bazel-bin/src/ovms_test --gtest_list_tests --gtest_filter="LLMChatTemplateTest.*:LLMOptionsHttpTest.*" | grep '^  '` ; do
+        ./bazel-bin/src/ovms_test --gtest_filter="*.$i" > tmp.log 2>&1 || ( failed=1 ; echo "TEST NAME $i" ; cat tmp.log ) 
+        cat tmp.log >> ${TEST_LOG}
```

Comment:
Could you add a comment why we break it down into two parts? Ideally this would be a temporary WA until we find the solution with multi initialization of Python interpreter in tests